uaclient/entitlements/repo.py

```diff
@@ -217,6 +215,16 @@ def install_packages(
         :param verbose: If true, print messages to stdout
         """
 
+        if not package_list:
+            package_list = self.packages
+
+        if not package_list:
+            return
+
+        msg_ops = self.messaging.get("pre_install", [])
+        if not util.handle_message_operations(msg_ops):
+            return
```

Comment:
Don't know if this is worth it, but we could change `install_packages` to return a boolean indicating if any packages was actually installed, then we can remove this message logic back into `_perform_enable`. What do you think ?

Reply:
That would work to print the message after installing, but this specifically needs to happen before installing as far as I can tell :/

Reply:
My bad here, I incorrectly read this as `post_install` :facepalm: